src/server/Document/Renderers/ComponentTrackingTemplate.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import processClientDeviceAndSendStaticBeacon from '#src/server/utilities/staticATITracking';
+import clickTracking from '#src/server/utilities/staticATITracking/clickTracking';
+import isOperaProxy from '#app/lib/utilities/isOperaProxy';
+import viewTracking from '#src/server/utilities/staticATITracking/viewTracking';
+
+const trackingScripts = ({
+  trackOnlyStaticCanonical = false,
+  trackComponentViews = true,
+} = {}) => {
+  return `
+    window.addEventListener('load', function () {
+      const isOperaMini = ${isOperaProxy.toString()}();
+      const isComponentTrackingEnabled = ${trackOnlyStaticCanonical} ? isOperaMini : true;
+      
+      if (!isComponentTrackingEnabled) {
+        return;
+      }
+
+      (${processClientDeviceAndSendStaticBeacon.toString()})();
+
+      if (${trackComponentViews}) {
+        (${viewTracking.toString()})();
+      }
+
+      (${clickTracking.toString()})();
```

Comment:
I wonder if we just implement the click + view tracking within a positive if statement:

```suggestion
      if (isComponentTrackingEnabled) {
          (${processClientDeviceAndSendStaticBeacon.toString()})();

          if (${trackComponentViews}) {
            (${viewTracking.toString()})();
          }

          (${clickTracking.toString()})(); 
      }
```